src/librustc_codegen_ssa/mir/place.rs

```diff
@@ -437,7 +437,7 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
         let result = match &place_ref {
```

Comment:
You could match on `(&place_ref.base, place_ref.projection.split_last())` here in order to simplify the later patterns